.github/workflows/deploy-container-image.yml

```diff
@@ -0,0 +1,63 @@
+# https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions
+name: Publish Base Container
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU for multi-arch builds
+        uses: docker/setup-qemu-action@v3
+        with:
+          # https://github.com/docker/setup-qemu-action/issues/198
+          # v9.2+ causes a segfault
+          image: tonistiigi/binfmt:qemu-v7.0.0-28
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/base
```

Comment:
I would prefer this structure:

For base container: `cachix/devenv:latest`

For devcontainer: `cachix/devenv/devcontainer:latest`

Consider it would be a breaking change for the devcontainer though

Reply:
Yeah, how about calling it `cachix/devenv/devenv`?

Reply:
You mean `cachix/devenv/devenv` for base devenv container and `cachix/devenv/devcontainer` for dev container?

because otherwise I think it would be even more confusing

Reply:
Yeah, so all new releases go to `devcontainer` for the main one.